vllm/sequence.py

```diff
@@ -250,8 +250,8 @@ def get_max_num_running_seqs(self) -> int:
                 # generation stage, we will have `best_of` sequences running.
                 return self.sampling_params.best_of
             # At sampling stages, return the number of actual sequences
-            # running.
-            return self.num_seqs(status=SequenceStatus.RUNNING)
+            # that are not finished yet.
+            return self.num_unfinished_seqs()
```

Comment:
The semantic of the function will change because this also counts swapped sequences. Is this expected?

Reply:
This is intended. `get_max_num_running_seqs` is being called for a swapped sequence when the swapped sequence is about to be swapped back to the running queue. This means these swapped sequences will soon become running sequences. This is the same logic as waiting sequences.